### PR TITLE
fix(tests): Handle failures during debian packaging builds

### PR DESCRIPTION
**Pull request by 3v1n0**
Fix some issues I got while building packages both locally (with sbuild, that as per commit 4a98eca runs as fakeroot) and in the PPA, mostly due to builders not being fast enough when spawning lots of parallel tests.

See failures: 
 - https://launchpad.net/~3v1n0/+archive/ubuntu/authd/+build/28517439
 - https://launchpad.net/~3v1n0/+archive/ubuntu/authd/+build/28517440

UDENG-3044

### CODE REVIEW

File: pam/integration-tests/cli_test.go [hunk 1]
```diff
@@ -63,6 +63,10 @@ func TestCLIAuthenticate(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			t.Parallel()
 
+			if !tc.currentUserNotRoot {
```
[the remark this answers is not in the document]
Comment (3v1n0) [reply pam/integration-tests/cli_test.go]: oh, yeah, sorry... It was a wrong paste, indeed it should follow the same logic.

Fixed.

**Comment by didrocks** [comment]
I am a little bit lost, shouldn’t `Rules-Requires-Root: no` that we have in debian/control do this for us, running as non root, so we don’t have to skip random tests?

**Comment by 3v1n0** [comment]
> Not a fan of increasing timeouts but I've been here before so I understand it's a necessary evil 😮‍💨

Yeah, I do agree... But well, at the end it's not a big deal

**Comment by 3v1n0** [comment]
> I am a little bit lost, shouldn’t `Rules-Requires-Root: no` that we have in debian/control do this for us, running as non root, so we don’t have to skip random tests?

Yeah, in fact it can be built without root... But the problem is when you build this using `sbuild` so locally using a clean environment (as I guess most of ubuntu developers do), where you do build as non-root but using fakeroot that for some reason is enough to break it.

However, as per the discussion we had in the previous related PR, the tests will be skipped only in this scenario, while for example they do work on launchpad (where schroot is still used, but with a configuration that prevents this to happen).

But let's say if you currently use the classical ubuntu developer build:
 - mk-sbuild noble
 - gbp buildpackage -S
 - sbuild -A -v -d noble-amd64 ../authd_*.dsc

We're currently failing, and it makes the process of debian package building and debugging very annoying.

**Comment by didrocks** [comment]
I don’t think this is a huge win to have something specialized for local sbuild, while remote build on LP is working. If that was the other way around, that would be a complete different story.

I wonder if we can’t rather look at the difference and see, if we can mimick another user in some other ways to make those tests meaningfull in every circumstances.

Not running the whole build + tests similar to the one on LP means that local builds are not really useful in the end to me and while we are at it, we could do ppa uploads… To let’s rather try to mirror the same runs between the 2.

**Comment by 3v1n0** [comment]
@didrocks I've dropped these commits for now, it will make authd test-packaging life harder, but I'll see if we can find a better way for it.

**Review by didrocks — APPROVED** [review]
Thanks for considering my feedback!

**Pull request merged**